File: 0617-merge-two-binary-trees/0617-merge-two-binary-trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {TreeNode}
 */
var mergeTrees = function(root1, root2) {
    function solve(root1, root2){
        if (root1==null){
            return root2
        }
        if (root2==null){
            return root1
        }
        const root= new TreeNode(root1.val + root2.val);
        root.left = solve(root1.left, root2.left)
        root.right = solve(root1.right, root2.right)
        return root
    }

    return solve(root1, root2)
    
};

File: 0617-merge-two-binary-trees/0617-merge-two-binary-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* mergeTrees(TreeNode* root1, TreeNode* root2) {
        TreeNode* result = new TreeNode(0);
        if (root1 == NULL && root2 == NULL) {
            result = NULL;
            return NULL;
        }
        if (root1 == NULL) {
            result = root2;
            return root2;
        }
        if (root2 == NULL) {
            result = root1;
            return root1;
        }
        result->val = root1->val + root2->val;
        result->left = mergeTrees(root1->left, root2->left);
        result->right = mergeTrees(root1->right, root2->right);
        return result;
    }
};